src/app/recommendations/recommendations.resolver.ts

```diff
@@ -34,6 +40,33 @@ export class RecommendationsResolver
     this.initialiseData();
   }
 
+  admins$: Observable<IAdmin[]> = this.store.select(
+    (state) => state.admins.items
+  );
+
+  private async initFiltersFormData() {
+    this.admins$
+      .pipe(
+        filter((data) => !!data),
```

Comment:
Is this a check for "not undefined"?

Reply:
Yes. The observable gets subscribed to twice, the first time it's undefined. Instead of checking the data is returned from the subscription, I filter it out in the pipe before, such that it only subscribes when there is data.